### PR TITLE
Refactoring on how to choose recipe with sampleRate values

### PR DESCRIPTION
**Pull request by allmywant**
Now maximum calls of getRandom is 3, when recipes has same sampleRate values.

### CODE REVIEW

File: lib/mojito.js
Comment (kingo55) [inline lib/mojito.js]: Are you sure we need this? Consider we have the following test:

1. Control: 0.1
2. Treatment 1: 0.3
3. Treatment 2: 0.3
4. Treatment 3: 0.3

Let's start with: random number = 0.95 and checked samples = 0

1. Checked samples 0.1, Pass
2. Checked samples 0.4, Pass
3. Checked samples 0.7, Pass
4. Checked samples 1, Chosen, Break loop

Assuming the order of recipes with same sample rate stays the same from the sort function, this should work, right?

CC @dapperdrop

File: lib/mojito.js
Comment (dapperdrop) [inline lib/mojito.js]: Just to tidy up the english used in our variables:

`samplesWasSet` >> `samplesSet`

**Review by dapperdrop — CHANGES_REQUESTED** [review]

File: lib/mojito.js
Comment (allmywant) [reply lib/mojito.js]: @dapperdrop 
Fixed.

**Review by dapperdrop — APPROVED** [review]

**Review by kingo55 — COMMENTED** [review]
One minor question here:

File: lib/mojito.js
Comment (allmywant) [reply lib/mojito.js]: @kingo55 
Yes, that should work. But the original thought on randomly choosing one recipe from recipes that has same sample rate was:

Just like the test you mentioned. 
With original random function (Math.random) and do not randomly choosing one recipe from recipes that has same sample rate, I remember you reported that the first recipe "Treatment 1"'s chosen rate much higher than "Treatment 1" and "Treatment 2", so we did that: if one recipe was chosen and there are other recipes that have same sample rate, then we randomly choose one them.

I can't remember more details, if we don't need this anymore with the SeededRNGRandom or MD5 hash, let's change it just like what you propoused.

Comment (kingo55) [reply lib/mojito.js]: Oh yeah, I remember now. 

It was such a long time ago that I forget why this was happening. My JS was not very good back then. If we make this change now, we'll probably have to validate this works out in the wild and that we're getting good, even assignment.

Comment (allmywant) [reply lib/mojito.js]: @kingo55 
Fixed, and I've taken some tests on it, it's working as expected.

Comment (kingo55) [reply lib/mojito.js]: @allmywant 
Oh great - I'll give it a test this end too.

**Comment by kingo55** [comment]
Also, would it be worth having another unit test that checks recipes with different and same sample rates, out of order reliably return a specific recipe?

E.g. We have 5 recipes:

1. 0.4
2. 0.1
3. 0.1
4. 0.2
5. 0.2

And we check that for `decision = 0.15` we always get recipe `3`, for `decision = 0.55` we always get recipe `4`, and for `decision = 0.65` we get recipe `1`.

Or is there a better way to check that decisions always yield the same recipes?

**Comment by allmywant** [comment]
@kingo55 
You're right, I will add a unit test to test this.
BTW: I think this pr might has confilcts with Sam's (#37 ), I can wait until Sam's pr was merged, then close this pr and I will create a new one based on Sam's changes.

**Comment by allmywant** [comment]
I've sent another PR, closing this one.

**Pull request closed without merging**
